base/promotion.jl

```diff
@@ -502,10 +502,12 @@ function TupleOrBottom(tt...)
 end
 
 """
-    promote_op(f, argtypes...)
+    promote_op(f, argtypes...; throw_error=false)
 
 Guess what an appropriate container eltype would be for storing results of
 `f(::argtypes...)`. The guess is in part based on type inference, so can change any time.
+If `f(::argtypes...)` does not return (e.g., a method does not exist) then `Union{}` will be returned.
```

Comment:
I could move this into the `Extended help` section if folks prefer.

Reply:
I think this is worth mentioning in the main text, it's brief.

```suggestion
If `f(::argtypes...)` does never return (e.g., a method does not exist) then `Union{}` may be returned.
```

`promote_op` may not be clever enough to determine that f does not return. In this case it would be valid for `promote_op` to return literally any type. Consequently will => may